File: week_2/subset.rb
require_relative 'randomized_queue'

class Subset
  def initialize(k, input)
    k = k.to_i
    queue = RandomizedQueue.new
    array = input.split(' ')

    array.each do |item|
      queue.enqueue(item)
    end

    k.times do
      puts queue.dequeue
    end
  end
end

input = $stdin.read
Subset.new(ARGV.first, input)

File: submissions/Subset.java
public class Subset {
    public static void main(String[] args) {
        int k = Integer.parseInt(args[0]);

        RandomizedQueue<String> queue = new RandomizedQueue<String>();
        for (int i = 0; i < k; i++) {
            String string = StdIn.readString();
            queue.enqueue(string);
        }

        for (int i = 0; i < k; i++) {
            System.out.println(queue.dequeue());
        }
        System.exit(0);
    }
}
